### PR TITLE
ci: bump cfn-lint to v1

### PR DESCRIPTION
**Pull request by zxkane**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/1/2024, 05:58:33 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1500&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1500&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1500&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/1/2024, 06:44:52 (UTC+0)_

File: src/ingestion-server/custom-resource/update-alb-rules.ts [hunk 1]
```diff
@@ -103,7 +107,7 @@ function createUpdateAlbRulesLambda(scope: Construct, listenerArn: string, input
     scope,
     'authEnableCondition',
     {
-      expression: Fn.conditionNot(Fn.conditionEquals(authenticationSecretArn, '')),
+      expression: Fn.conditionEquals(enableAuthentication, 'Yes'),
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules.ts:110]: use enum FeatureFlag?

Comment (dengmingtong) [reply src/ingestion-server/custom-resource/update-alb-rules.ts:110]: This part is CDK code for template.

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/1/2024, 09:06:00 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/2/2024, 03:44:57 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/2/2024, 09:52:03 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/2/2024, 13:15:52 (UTC+0)_

File: .github/workflows/dist-scan.yml [hunk 2]
```diff
@@ -56,15 +56,16 @@ jobs:
...
+          cfn-lint -i W3005 W2001 W3045 W8003 E1010 E1022 -e -r cn-north-1,cn-northwest-1 --ignore-templates $TEMPLATE_ROOT/data-reporting-quicksight-stack.template.json --ignore-templates $TEMPLATE_ROOT/*NewServerlessRedshift*.nested.template.json --ignore-templates $TEMPLATE_ROOT/data-pipeline-stack.template.json --ignore-templates $TEMPLATE_ROOT/datapipeline*.nested.template.json --ignore-templates $TEMPLATE_ROOT/cloudfront-s3-control-plane-stack-global*.json --ignore-templates $TEMPLATE_ROOT/*cognito-control-plane-stack.template.json --ignore-templates $TEMPLATE_ROOT/public-exist-vpc-custom-domain-control-plane-stack.template.json --ignore-templates $TEMPLATE_ROOT/ingestionserver*.nested.template.json  -t $TEMPLATE_ROOT/*.template.json
```
Comment (zxkane) [inline .github/workflows/dist-scan.yml:68]: why suppress those two rules when checking templates  in gcr?

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1500)
###### _updated: 7/3/2024, 06:41:04 (UTC+0)_

**Review by amliuyong — APPROVED** [review]

**Pull request merged**